### DISCUSSION TITLE
understanding output of brew test-bot and troubleshooting github actions

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#2489] [Tap maintenance and Homebrew development]
been working on getting a valid bottle / working for `freecad.rb` formula going for the homebrew-freecad tap.

freecad has no shortage of dependencies that is relies on. so when i have to wait ~ 2 hours for a build to go through the pipeline, and then fail, only see something like the below, it becomes quite daunting to track down 🕵️‍♀️ what exactly is causing the issue.

https://github.com/FreeCAD/homebrew-freecad/runs/4255635149?check_suite_focus=true#step:13:30

```
==> FAILED
Full audit freecad/freecad/freecad --online --git --skip-style output
  Broken dependencies:
    /usr/local/opt/hdf5@1.10/lib/libhdf5.103.dylib (hdf5@1.10)
  freecad/freecad/freecad:
    * freecad has broken dynamic library links:
        
  Error: 1 problem in 1 formula detected
  Error: freecad has broken dynamic library links:
    
  
  ::endgroup
  
  ==> brew bottle --verbose --json freecad/freecad/freecad --root-url=FreeCAD/homebrew-freecad/releases/download/freecad-0.19
  ==> brew bottle --merge --write --no-commit ./freecad--0.19.catalina.bottle.json
  ==> brew uninstall --force freecad/freecad/freecad
  ==> brew uninstall --force autoconf autoconf@2.69 automake bison cmake cython docbook docbook-xsl freecad/freecad/swig@4.0.2 hdf5@1.10 libtool m4 meson ninja openjdk pyqt-builder python@3.10 rapidjson scons sip swig util-macros xcb-proto xtrans
  ==> brew install --only-dependencies ./freecad--0.19.catalina.bottle.1.tar.gz
  ==> brew install ./freecad--0.19.catalina.bottle.1.tar.gz
  ==> brew linkage --test freecad/freecad/freecad
  ==> FAILED
Full linkage --test freecad/freecad/freecad output
  Broken dependencies:
    /usr/local/opt/hdf5@1.10/lib/libhdf5.103.dylib (hdf5@1.10)
  ::endgroup
Truncated linkage --test freecad/freecad/freecad output
  Warning: Broken dependencies:
  
    /usr/local/opt/hdf5@1.10/lib/libhdf5.103.dylib (hdf5@1.10)
  
  ::endgroup
```

what is a decent way to work through these error messages. as running the command locally on my catalina box doesn't produce the same error so it leads to a lot of hunt-and-peck, and waiting ~ 2 hours for each little incremental change i apply to formula makes this whole process time consuming to say the least.

any and all thoughts and suggestions are welcome. 🙏

**Comment by Bo98** [comment]
Looks like `hdf5@1.10` is a runtime dependency in terms of linkage, but it is only a build-time and test-time dependency in your dep tree.

You would get the same issue locally if you were to `brew uninstall hdf5@1.10`.

**Reply by ipatch** [reply]
thanks. i'll experiment with this, and see if it can bump along to the next error (if there is one).
